tree/ntuple/v7/inc/ROOT/RColumnElement.hxx

```diff
@@ -27,6 +28,54 @@
 #include <string>
 #include <type_traits>
 
+#ifndef R__LITTLE_ENDIAN
+#if defined(_WIN32) || defined(__LITTLE_ENDIAN__) || \
+  (defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+# define R__LITTLE_ENDIAN 1
+#elif defined(__BIG_ENDIAN__) || \
+  (defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+# define R__LITTLE_ENDIAN 0
+#else
+# error "Unsupported endianness"
+#endif
+#endif /* R__LITTLE_ENDIAN */
+
+namespace {
+template <std::size_t N>
+struct RByteSwap {};
+
+template <>
+struct RByteSwap<2> {
+   using value_type = std::uint16_t;
+   static value_type bswap(value_type x) { return Rbswap_16(x); }
+};
+
+template <>
+struct RByteSwap<4> {
+   using value_type = std::uint32_t;
+   static value_type bswap(value_type x) { return Rbswap_32(x); }
+};
+
+template <>
+struct RByteSwap<8> {
+   using value_type = std::uint64_t;
+   static value_type bswap(value_type x) { return Rbswap_64(x); }
+};
+
+/// \brief Copy and byteswap `count` elements of size `N` from `source` to `destination`.
+///
+/// Used on big-endian architectures for packing/unpacking elements whose column type requires
+/// a little-endian on-disk representation.
+template <std::size_t N>
+static void CopyElementsBswap(void *destination, const void *source, std::size_t count) {
+   auto dst = reinterpret_cast<typename RByteSwap<N>::value_type *>(destination);
+   auto src = reinterpret_cast<const typename RByteSwap<N>::value_type *>(source);
+   for (std::size_t i = 0; i < count; ++i) {
```

Comment:
Perhaps the compiler optimizes it away but it looks like every iteration needs to increment three variables (`i`, `src`, `dst`). Wouldn't it be faster to use `i` as an array index to `src` and `dst`?

Reply:
Well, `src[i]` is just syntactic sugar for `*(src + i)`, and `src + i` implies (depending on the architecture addressing modes), a multiplication; what is faster depends on the compiler and target architecture.
Anyways, if there is a difference, it should be negligible, so I will change to `src[i]`, which is much more readable.  :slightly_smiling_face: